examples/search_with_specific_engine.py

```diff
@@ -4,21 +4,37 @@
 """
 import asyncio
 
+from metagpt.config2 import Config
 from metagpt.roles import Searcher
 from metagpt.tools.search_engine import SearchEngine, SearchEngineType
 
 
 async def main():
     question = "What are the most interesting human facts?"
-    kwargs = {"api_key": "", "cse_id": "", "proxy": None}
-    # Serper API
-    # await Searcher(search_engine=SearchEngine(engine=SearchEngineType.SERPER_GOOGLE, **kwargs)).run(question)
-    # SerpAPI
-    # await Searcher(search_engine=SearchEngine(engine=SearchEngineType.SERPAPI_GOOGLE, **kwargs)).run(question)
-    # Google API
-    # await Searcher(search_engine=SearchEngine(engine=SearchEngineType.DIRECT_GOOGLE, **kwargs)).run(question)
-    # DDG API
-    await Searcher(search_engine=SearchEngine(engine=SearchEngineType.DUCK_DUCK_GO, **kwargs)).run(question)
+
+    search = Config.default().search
+    kwargs = {"api_key": search.api_key, "cse_id": search.cse_id, "proxy": None}
+
+    if search.api_type == SearchEngineType.DIRECT_GOOGLE:
+        # Google API
+        await Searcher(
+            search_engine=SearchEngine(engine=SearchEngineType.DIRECT_GOOGLE, **kwargs)
+        ).run(question)
+    elif search.api_type == SearchEngineType.SERPER_GOOGLE:
```

Comment:
You can replace these if-else statements with a dict to reduce complexity, like this:
```python
mappings = {
    SearchEngineType.DIRECT_GOOGLE: {
        "api_key": search.api_key, 
        "cse_id": search.cse_id, 
        "proxy": None, 
        "engine":SearchEngineType.DIRECT_GOOGLE}, 
     ...}
args = mappings.get(search.api_type)
search_engine=SearchEngine(**args)
```

Reply:
search = Config.default().search
kwargs = {"api_key": search.api_key, "cse_id": search.cse_id, "proxy": None}await Searcher(search_engine=SearchEngine(engine=search.api_type, **kwargs)).run(question)
 
    It only needs this right when you just made this suggestion I looked into this

Reply:
`examples/search_with_specific_engine.py` after running pre-commit:
```python
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
"""
import asyncio

from metagpt.config2 import Config
from metagpt.roles import Searcher
from metagpt.tools.search_engine import SearchEngine, SearchEngineType


async def main():
    question = "What are the most interesting human facts?"

    search = Config.default().search
    kwargs = {"api_key": search.api_key, "cse_id": search.cse_id, "proxy": None}

    if search.api_type == SearchEngineType.DIRECT_GOOGLE:
        # Google API
        await Searcher(search_engine=SearchEngine(engine=SearchEngineType.DIRECT_GOOGLE, **kwargs)).run(question)
    elif search.api_type == SearchEngineType.SERPER_GOOGLE:
        # Serper API
        await Searcher(search_engine=SearchEngine(engine=SearchEngineType.SERPER_GOOGLE, **kwargs)).run(question)
    elif search.api_type == SearchEngineType.SERPAPI_GOOGLE:
        # SerpAPI
        await Searcher(search_engine=SearchEngine(engine=SearchEngineType.SERPAPI_GOOGLE, **kwargs)).run(question)
    else:
        # DDG API
        await Searcher(search_engine=SearchEngine(engine=SearchEngineType.DUCK_DUCK_GO, **kwargs)).run(question)


if __name__ == "__main__":
    asyncio.run(main())
```

Reply:
This is the cyclomatic complexity after cascading if-else statements:
```shell
radon cc -a examples/search_with_specific_engine.py
examples/search_with_specific_engine.py
    F 12:0 main - A

1 blocks (classes, functions, methods) analyzed.
Average complexity: A (4.0)
```
In the case of this code, the cyclomatic complexity is acceptable.

Reply:
This is the latest complexity showing when i have made(latest commit on 1 the above mentioned change

radon cc -a search_with_specific_engine.py
search_with_specific_engine.py
    F 12:0 main - A

1 blocks (classes, functions, methods) analyzed.
Average complexity: A (1.0)